### DISCUSSION TITLE
Cache usage from swr import

### DISCUSSION THREAD

**Discussion by Meyclem** [vercel/swr#1294] [Q&A] [answered]
Hi!

It is actually possible to do something like that:
```js
import { cache } from "swr"

cache.set(key, value)
```

The problem is I can't see any documentation about that, except in the beta swr version, which I don't use and me I my team don't want to use something that isn't documented as it would mean that it's not the good way to do things.

Do I miss something here?

**Comment by huozhi** [comment, marked as answer]
The exported cache is experimental feature in swr 0.x, which is not encouraged to use. It will be removed in swr 1.x.
You can try the cache API in swr beta version, checkout https://swr.vercel.app/docs/advanced/cache

Feel free to report encountered issues or submit any feature request for it

**Reply by Meyclem** [reply]
Thanks for the answer.

So, if I understand correctly, the way to get the value of the cache for a given key is to call the endpoint again (which results in a 304 response)?

**Reply by huozhi** [reply]
That will depends on you, we'd expect user can choose the way to revalidate or not.
We'll announce a new option `revalidateWhenStale` in v1 to give user ability to make their own decision. You can either let fetcher run again or just keep use the stale cache.

**Reply by nandorojo** [reply]
@huozhi I notice the docs for the new cache only allow for string keys. Is this a breaking change for array keys?

**Reply by shuding** [reply]
@nandorojo Nope. Array as the key is still supported but there's internal serialization under the hood to ensure the actual cache keys are strings. With that the cache provider itself doesn't need to care about those case, it just need to work like a normal KV.
